cmd/daemon.go

```diff
@@ -15,14 +21,62 @@ func NewDaemonCmd(global *globalFlags) *cobra.Command {
 		Use:   "daemon",
 		Short: "Run the crypta daemon",
 		RunE: func(cmd *cobra.Command, args []string) error {
-			return c.Run(args)
+			if !c.detached {
+				return cmd.Help()
+			}
+
+			return c.run()
+		},
+	}
+
+	start := &cobra.Command{
+		Use:   "start",
+		Short: "Starts the crypta daemon",
+		RunE: func(cmd *cobra.Command, args []string) error {
+			return c.start(args)
+		},
+	}
+
+	stop := &cobra.Command{
+		Use:   "stop",
+		Short: "Stops the crypta daemon",
+		RunE: func(cmd *cobra.Command, args []string) error {
+			return c.stop()
 		},
 	}
 
+	cc.AddCommand(start)
+	cc.AddCommand(stop)
+
+	cc.Flags().BoolVar(&c.detached, "detached", false, "Runs the daemon detached")
+	_ = cc.Flags().MarkHidden("detached")
+
 	return cc
 }
 
-func (c *daemonCmd) Run(args []string) error {
+func (c *daemonCmd) start(args []string) error {
```

Comment:
I would not combine these three independent entry points in the `daemonCmd` struct. Instead I would split it up by responsibility: one for running, one for starting and one for stopping the daemon. This would also allow us to have seperate flagsets for each subcommand.

Reply:
So you would propose to introduce 3 structs for the 3 entry points?

I am not sure about this. If we would like to change the flagset in the future, we could still do this change. Until then, I would propose to follow `YAGNI` and keep it as simple as possible.

I currently don't see any need for different flagsets for the three "commands". Do you?

Reply:
Yes I would keep the structure consistent and have one flagset and one entrypoint per struct like for the other commands. In this case the struct is a container for a command logic with its flags. Since these logics and the corresponding flags can change indepedently for different reasons at different times, I would stick to the SRP and separate them.

Regarding the flagsets I agree. In the future it could be that some arguments for the automatic stopping of the daemon could be added to `start` and a timeout for killing it the to `stop` command. For now I would also keep it as simple as possible.